diff --git a/node_modules/remix-auth-spotify/build/index.js b/node_modules/remix-auth-spotify/build/index.js
index 087057b..d7ced8c 100644
--- a/node_modules/remix-auth-spotify/build/index.js
+++ b/node_modules/remix-auth-spotify/build/index.js
@@ -30,18 +30,26 @@ class SpotifyStrategy extends remix_auth_oauth2_1.OAuth2Strategy {
                 Authorization: `Bearer ${accessToken}`,
             },
         });
-        const data = await response.json();
-        const profile = {
-            provider: 'spotify',
-            displayName: data.display_name,
-            id: data.id,
-            emails: [{ value: data.email }],
-            photos: data.images.length > 0
-                ? [{ value: data.images[0].url }]
-                : undefined,
-            __json: data,
-        };
-        return profile;
+        try {
+            const data = await response.json();
+            const profile = {
+                provider: 'spotify',
+                displayName: data.display_name,
+                id: data.id,
+                emails: [{ value: data.email }],
+                photos: data.images.length > 0
+                    ? [{ value: data.images[0].url }]
+                    : undefined,
+                __json: data,
+            };
+            return profile;
+        } catch (e) {
+            const options = {
+                sessionKey: this.sessionKey,
+                sessionErrorKey: this.sessionErrorKey,
+            };
+            return this.handleResult(null, options, 'Forbidden by Spotify', true);
+        }
     }
     async getAccessToken(response) {
         const { access_token: accessToken, token_type: tokenType, expires_in: expiresIn, refresh_token: refreshToken, } = await response.json();
